vslm/vslm_object_manager.go

```diff
@@ -0,0 +1,237 @@
+package vslm
+
+import (
+	"context"
+
+	//"github.com/vmware/govmomi/vim25"
+	//	vim "github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/object"
+	vim "github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/vslm/methods"
+	"github.com/vmware/govmomi/vslm/types"
+)
+
+type VslmTask struct {
+	c *Client
+	object.Common
+	vim.ManagedObjectReference
+}
+
+func NewVslmTask(client *Client, mref vim.ManagedObjectReference) *VslmTask {
+	m := VslmTask{
+		ManagedObjectReference: mref,
+		c:                      client,
+	}
+	return &m
+}
+
+func (t *VslmTask) QueryResult(ctx context.Context) (vim.AnyType, error) {
+	req := types.VslmQueryTaskResult{}
+	res, err := methods.VslmQueryTaskResult(ctx, t.c, &req)
+	if err != nil {
+		return nil, err
+	}
+	return &res.Returnval, nil
+}
+
+func (t *VslmTask) QueryInfo(ctx context.Context) (*types.VslmTaskInfo, error) {
+	req := types.VslmQueryInfo{}
+	res, err := methods.VslmQueryInfo(ctx, t.c, &req)
+	if err != nil {
+		return nil, err
+	}
+	return &res.Returnval, nil
+}
+
+func (t *VslmTask) Cancel(ctx context.Context) error {
+	req := types.VslmCancelTask{}
+	_, err := methods.VslmCancelTask(ctx, t.c, &req)
+	return err
+}
+
+type VslmObjectManager struct {
```

Comment:
Maybe we could name this `GlobalObjectManager` instead?  go-lint complains when package and type prefix are the same.